openassessment/xblock/static/js/src/studio/oa_edit_rubric.js

```diff
@@ -37,6 +35,71 @@ export class EditRubricView {
       },
     );
     this.criteriaContainer.addEventListeners();
+    this.setupRubricReuse();
+  }
+
+  /**
+   * Sets up the UI for rubric cloning.
+   *
+   * Setup JQuery-UI Accordion dropdown panel, autocomplete input, autocomplete/combobox button.
+   * Load json representation of other course ORAs from rendered JSON
+   * Install click handler on the "Clone" button to submit request and display success banner.
+   */
+  setupRubricReuse() {
+    const view = this;
+    // If the section hasn't been rendered, rubric reuse isn't enabled, so return early
+    if ($('#openassessment_rubric_clone').size() === 0) {
+      return;
+    }
+    // Set up the accordion dropdown section
+    $('#openassessment_rubric_clone', view.element).accordion({
+      collapsible: true,
+      heightStyle: 'content',
+      icons: {
+        header: false,
+        activeHeader: false,
+      },
+    });
+    // Load the other_ora_data for the autocomplete
+    const otherOraData = $.parseJSON(
+      $('#openassessment_rubric_clone_other_ora_data').text(),
+    ).map((otherOra) => (otherOra.location));
+
+    // Setup the Autocomplete on the text input
+    this.rubricAutocomplete = $('#openassessment_rubric_clone_input').autocomplete({
+      delay: 0,
+      minLength: 0,
+      source: otherOraData,
+    });
+
+    // Setup the 'combobox' dropdown button. Based on https://jqueryui.com/autocomplete/#combobox
+    let wasOpen = false;
+    $('.openassessment_clone_combobox_toggle')
+      .button()
+      .removeClass('ui-corner-all')
+      .addClass('custom-combobox-toggle ui-corner-right')
+      .on('mousedown', () => {
```

Comment:
so... grain of salt... but I feel like these callbacks should all just be other bound methods of the class rather than in-line paritials.

Grain of salt because this is not react code, and is not going to live for super-duper-long, but that version is easier to maintain and test, and will be easier to steal pieces of later when we attempt the final refactor.

There is a small chance that I'm missing an eccentricity about the way you are using `view=this` because there *are* specific instances where jquery kind of requires that, but we don't *seem* to be hitting any of those in this code.